# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/src/peerconnection/RemoteTrackSource.h
# Commit: c54cb3c10992
# Full Hash: c54cb3c109922bc15fb6d0ca67e5f2a05980c1a5
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-31 15:46:43
# Regressor Bug: 1493613
# File Overlap Count: 2
# Description:
#   Bug 1493613 - Move MediaStream control from DOMMediaStream to MediaStreamTrack. r=padenot
#   
#   This is inherently large, because modifying these bits of DOMMediaStream and
#   MediaStreamTrack affects all consumers and producers of all DOMMediaStreams and
#   MediaStreamTracks.
# ==============================================================================

diff -r 0b03dd9d20ac -r c54cb3c10992 media/webrtc/signaling/src/peerconnection/RemoteTrackSource.h
--- a/media/webrtc/signaling/src/peerconnection/RemoteTrackSource.h	Wed Jul 31 09:35:06 2019 +0000
+++ b/media/webrtc/signaling/src/peerconnection/RemoteTrackSource.h	Wed Jul 31 07:58:17 2019 +0000
@@ -12,8 +12,9 @@
 
 class RemoteTrackSource : public dom::MediaStreamTrackSource {
  public:
-  explicit RemoteTrackSource(nsIPrincipal* aPrincipal, const nsString& aLabel)
-      : dom::MediaStreamTrackSource(aPrincipal, aLabel) {}
+  explicit RemoteTrackSource(SourceMediaStream* aStream,
+                             nsIPrincipal* aPrincipal, const nsString& aLabel)
+      : dom::MediaStreamTrackSource(aPrincipal, aLabel), mStream(aStream) {}
 
   dom::MediaSourceEnum GetMediaSource() const override {
     return dom::MediaSourceEnum::Other;
@@ -43,9 +44,15 @@
     PrincipalChanged();
   }
   void SetMuted(bool aMuted) { MutedChanged(aMuted); }
+  void ForceEnded() { OverrideEnded(); }
+
+  const RefPtr<SourceMediaStream> mStream;
 
  protected:
-  virtual ~RemoteTrackSource() {}
+  virtual ~RemoteTrackSource() {
+    MOZ_ASSERT(NS_IsMainThread());
+    MOZ_ASSERT(mStream->IsDestroyed());
+  }
 };
 
 }  // namespace mozilla